core/src/main/java/jenkins/model/GlobalCloudConfiguration.java

```diff
@@ -1,30 +1,56 @@
 package jenkins.model;
 
 import hudson.Extension;
+import hudson.RestrictedSince;
+import hudson.model.Descriptor;
+import hudson.model.RootAction;
 import hudson.slaves.Cloud;
+import hudson.util.FormApply;
 import net.sf.json.JSONObject;
 import org.jenkinsci.Symbol;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
 import org.kohsuke.stapler.StaplerRequest;
+import org.kohsuke.stapler.StaplerResponse;
+import org.kohsuke.stapler.verb.POST;
 
+import javax.annotation.CheckForNull;
+import javax.servlet.ServletException;
 import java.io.IOException;
 
 /**
- * Adds the {@link Cloud} configuration to the system config page.
+ * Provides a configuration form for {@link Jenkins#clouds}.
  *
- * <p>
- * This object just acts as a proxy to configure {@link Jenkins#clouds}
- *
- * @author Kohsuke Kawaguchi
+ * Has been overhauled in Jenkins 2.XXX to no longer contribute to Configure System, but be a standalone form.
  */
-@Extension(ordinal=-100) @Symbol("cloud") // historically this was placed at the very end of the configuration page
-public class GlobalCloudConfiguration  extends GlobalConfiguration {
+@Extension
+@Symbol("cloud")
+@Restricted(NoExternalUse.class)
```

Comment:
This would prohibit JCasC from configuring it by default

Reply:
JCasC ignores it on the class level, only looks at it on getters afaik

Reply:
See tested example where it works:
https://github.com/jenkinsci/configuration-as-code-plugin/blob/3a67b90dec1ccc0aece636836998f137f526af2a/plugin/src/main/java/io/jenkins/plugins/casc/core/ProxyConfigurator.java#L72

Reply:
@timja wrote on [Sunday](https://github.com/jenkinsci/jenkins/pull/4339#pullrequestreview-314617924):

> Tested manually with JCasC + the azure-vm agents plugin, all fine

So please provide some supporting evidence for your claim.

Reply:
OK. I'd guess it is rather a subject for the future when JCasC starts consulting with class- and package- level annotations for such use-cases. Does not block it now